### PR TITLE
Integrate DynamoDB and Update Dependencies in Ivy Chatbot Application

### PR DESCRIPTION
**Pull request by gracebrazil28**
Requirements.txt Updates:

1. Added boto3 to enable communication with DynamoDB.

Main.py Updates:

1. Added functions for logging user login and chat history to DynamoDB.
2. Commented out CSV logging code while retaining the structure.
3. Ensured AWS credentials are correctly loaded from environment variables.
4. Implemented logging and updating of chat history in DynamoDB.
5. Updated reaction column to reflect user interactions (e.g., 'liked' or 'disliked').
6. Used username and timestamp as primary keys to handle updates in the chat history table.
7. Changed timestamp format from int Unix timestamp to a more readable time and date string.
8. Added log_user_login function to record user logins in anticipation of a login interface.

Created test_dynamodb_connection.py:
1. Added a testing script to verify DynamoDB setup and interaction with the web app.

### CODE REVIEW

**Review by cs1090218 — APPROVED** [review]
Could you change the table names and column names to follow the upper camel case formatting as mentioned here: https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/AppendixSampleTables.html?
Other than that it's good.

**Comment by Rochan-HM** [comment]
Looks good.
I tried creating a deployment of the PR here: https://dashboard.dilab-ivy.com/project/vkowksg/production/application/o4s8kw4/deployment/zwwg4oo

Looks like it's downloading tons of versions of the AWS SDK.
```
Downloading botocore-1.26.7-py3-none-any.whl (8.8 MB)
Downloading botocore-1.26.6-py3-none-any.whl (8.8 MB)
```
We might run out of memory building this image.

Looks like it is an issue with PIP unfortunately: https://github.com/pypa/pip/issues/8713

Can we specify the exact version in our requirements file?

**Comment by gracebrazil28** [comment]
Thanks for the review guys! New changes according to your suggestions in Commit 3 SHA: 06af3c7a839e4747736a90517e8f5ad1c450b1de 
@Rochan-HM  I added the botocore and boto3 versions
@cs1090218  I edited table names and column names

File: main.py [hunk 1]
```diff
@@ -140,10 +141,10 @@ def update_chat_history(user_id, session_id, question, response, reaction):
         # Update item in DynamoDB
         response = chat_history_table.update_item(
             Key={
-                'username': user_id,
-                'timestamp': formatted_timestamp,
+                'Username': user_id,
+                'Timestamp': formatted_timestamp,
             },
-            UpdateExpression='SET reaction = :reaction',
```
Comment (cs1090218) [inline main.py]: Should this be 'SET Reaction = :reaction' with a capital R in first reaction?

**Comment by Rochan-HM** [comment]
LGTM!
Just tested it out and it works: https://1no00c8o.dev.dilab-ivy.com/

Minor: Looks like we have an error with the `gr.Info` component:
image
Let me push an update to fix that.

**Review by Rochan-HM — APPROVED** [review]

**Comment by gracebrazil28** [comment]
@Rochan-HM Thanks for fixing the invalid kwarg from the pip version changes. Looks good to me. Thank you both!

**Pull request merged**